test/refda/test_eval.c

```diff
@@ -380,46 +380,138 @@ TEST_CASE("821183820402820402840101256A636F6D706172652D6765", "8201F5")
 TEST_CASE("8211838201F4F78401012568626F6F6C2D616E64", "F7")
 // ari:/AC/(/INT/0x0,/UVAST/1,//1/1/OPER/divide) -> undefined
 TEST_CASE("8211838204008207018401012566646976696465", "F7")
 // ari:/AC/(/INT/0x0,/UVAST/1,//1/1/OPER/remainder) -> undefined
 TEST_CASE("821183820400820701840101256972656D61696E646572", "F7")
-void test_refda_eval_target_undefined(const char *targethex, const char *expectloghex)
+
+// Test addition of timespec related logic:
+//
+// --- Addition of TP and TD value (in either order) ---
+// ari:/AC/(/TD/1000,/TP/20000101T001640Z,//1/1/OPER/add) -> /TP/20000101T003320Z
+TEST_CASE("821183820D1903E8820C1903E88401012563616464", "820C1907D0")
+// ari:/AC/(/TP/20000101T001640Z,/TD/1000,//1/1/OPER/add) -> /TP/20000101T003320Z
+TEST_CASE("821183820C1903E8820D1903E88401012563616464", "820C1907D0")
+//
+// --- Addition of 2 TD values ---
+// ari:/AC/(/TD/5000,/TD/7000,//1/1/OPER/add) -> /TD/12000
+TEST_CASE("821183820D191388820D191B588401012563616464", "820D192EE0")
+// ari:/AC/(/TD/7000,/TD/5000,//1/1/OPER/add) -> /TD/12000
+TEST_CASE("821183820D191B58820D1913888401012563616464", "820D192EE0")
+// ari:/AC/(/TD/4.75,/TD/2.75,//1/1/OPER/add) -> /TD/7.5
+TEST_CASE("821183820D82211901DB820D82211901138401012563616464", "820D8220184B")
+
+// Test subtraction of timespec related logic:
+//
+// --- Subtraction of 2 TP values ---
+// ari:/AC/(/TP/20000101T001640Z,/TP/20000101T001640Z,//1/1/OPER/sub) -> /TD/0
+TEST_CASE("821183820C1903E8820C1903E88401012563737562", "820D00")
+// ari:/AC/(/TP/20000101T001600Z,/TP/20000101T001640Z,//1/1/OPER/sub) -> /TD/40
+TEST_CASE("821183820C1903C0820C1903E88401012563737562", "820D1828")
+// ari:/AC/(/TP/20000101T001640Z,/TP/20000101T001600Z,//1/1/OPER/sub) -> /TD/-40
+TEST_CASE("821183820C1903E8820C1903C08401012563737562", "820D3827")
+//
+// --- Subtraction of 2 TD values ---
+// ari:/AC/(/TD/4000,/TD/7000,//1/1/OPER/sub) -> /TD/3000
+TEST_CASE("821183820D190FA0820D191B588401012563737562", "820D190BB8")
+// ari:/AC/(/TD/7000,/TD/4000,//1/1/OPER/sub) -> /TD/-3000
+TEST_CASE("821183820D191B58820D190FA08401012563737562", "820D390BB7")
+// ari:/AC/(/TD/0.25,/TD/3.75,//1/1/OPER/sub) -> /TD/3.5
+TEST_CASE("821183820D82211819820D82211901778401012563737562", "820D82201823")
+//
+//-----------------------------------------------------------------------------------------------------------
+// TODO: The 2 unit tests below fail and it may be due to bad parsing of negative fractional TD values.
+// // ari:/AC/(/TD/-2.25,/TD/0.75,//1/1/OPER/sub) -> /TD/3
+// TEST_CASE("821183820D822138AE820D8221184B8401012563737562", "820D03")
+// // ari:/AC/(/TD/3.25,/TD/0.75,//1/1/OPER/sub) -> /TD/-2.5
+// TEST_CASE("821183820D8221190145820D8221184B8401012563737562", "820D82202E")
+//
+//-----------------------------------------------------------------------------------------------------------
+// TODO: Based on ticket #132, this should be an error. Should this eventually be supported?
```

Comment:
Good catch! I think it should, since as you pointed out subtraction is just addition with a different sign. The case seems valid to me.